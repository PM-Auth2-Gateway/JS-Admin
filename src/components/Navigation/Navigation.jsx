import React from 'react';
import classNames from 'classnames';
import { Image, Nav, Navbar } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import LoginButton from './../LoginButton/LoginButton';

import styles from './Navigation.module.scss';

const Navigation = () => {
  return (
    <Navbar collapseOnSelect expand='md' className={styles.border}>
      <NavLink to='/'>
        <Navbar.Brand>
          <Image
            src='/favicon.ico'
            alt='icon'
            width='30'
            height='30'
            className={classNames('mr-2', 'align-center', 'align-top')}
            rounded
          />
        </Navbar.Brand>
      </NavLink>
      <Navbar.Toggle aria-controls='responsive-navbar-nav' />
      <Navbar.Collapse id='responsive-navbar-nav'>
        <Nav className='mr-auto'>
          <Nav.Link
            as={NavLink}
            to='/applications'
            activeClassName='text-primary'
          >
            Applications
          </Nav.Link>
          <Nav.Link as={NavLink} to='/guide' activeClassName='text-primary'>
            Guide
          </Nav.Link>
          <Nav.Link as={NavLink} to='/docs' activeClassName='text-primary'>
            Docs
          </Nav.Link>
        </Nav>
        <LoginButton />
      </Navbar.Collapse>
    </Navbar>
  );
};

export default Navigation;
