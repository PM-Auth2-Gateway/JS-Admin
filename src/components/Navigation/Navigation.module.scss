.border{
  border-bottom: 1px solid rgb(238, 238, 238);
}

.dropdown{
  right: auto;
}