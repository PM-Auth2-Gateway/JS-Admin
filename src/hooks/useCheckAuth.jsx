import { useSelector } from 'react-redux';
import LocalStorageService from '../services/LocalStorageService';
import { createSelector } from 'reselect';

const isAuthenticatedSelector = createSelector(
  (state) => state.user,
  ({ authenticated }) => authenticated
);

const useCheckAuth = () => {
  const isAuth = useSelector(isAuthenticatedSelector);

  if (isAuth || LocalStorageService.getUser()) {
    return true;
  } else {
    return false;
  }
};

export default useCheckAuth;
