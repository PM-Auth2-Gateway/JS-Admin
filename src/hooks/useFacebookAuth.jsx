import { useEffect, useState } from 'react';

export const useFacebookAuth = () => {
  const [user, setUser] = useState(null);
  const [authenticated, setAuthenticated] = useState(false);
  useEffect(() => {
    window.fbAsyncInit = function () {
      window.FB.init({
        appId: '839201246938380',
        autoLogAppEvents: true,
        xfbml: true,
        version: 'v10.0',
      });
    };
    (function (d, s, id) {
      var js,
        fjs = d.getElementsByTagName(s)[0];
      if (d.getElementById(id)) return;
      js = d.createElement(s);
      js.id = id;
      js.src = '//connect.facebook.net/en_US/sdk.js';
      fjs.parentNode.insertBefore(js, fjs);
    })(document, 'script', 'facebook-jssdk');
  }, []);

  const login = () => {
    window.FB.login(() => {
      window.FB.api('/me?fields=id,name,email,picture', (user) => {
        setUser(user);
        setAuthenticated(true);
      });
    });
  };

  const logout = () => {
    window.FB.logout(() => {
      setAuthenticated(false);
    });
  };

  return { login, user, authenticated, logout };
};
