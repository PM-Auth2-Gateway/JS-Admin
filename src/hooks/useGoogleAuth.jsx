import { useEffect } from 'react';
const useGoogleAuth = () => {
  useEffect(() => {
    window.gapi.load('auth2', () => {
      window.gapi.auth2.init({
        clientId:
          '792255729562-tjcf5k4uv35et7e3v0e1g2pkmqibpsn8.apps.googleusercontent.com',
      });
    });
  }, []);

  const login = () => {
    return window.gapi.auth2
      .getAuthInstance()
      .signIn()
      .then((user) => {
        const profile = user.getBasicProfile();
        const userData = {
          name: profile.getName(),
          email: profile.getEmail(),
          picture: profile.getImageUrl(),
        };
        return userData;
      })
      .then((res) => res);
  };

  const logout = () => {
    return window.gapi.auth2.getAuthInstance().signOut();
  };

  return { loginWithGoogle: login, logoutWithGoogle: logout };
};

export default useGoogleAuth;
