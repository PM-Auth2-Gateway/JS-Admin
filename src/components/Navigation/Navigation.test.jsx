import { shallow } from 'enzyme';
import Navigation from './Navigation';
import { useFacebookAuth } from './../../hooks/useFacebookAuth';

jest.mock('./../../hooks/useFacebookAuth', () => ({
  ...jest.requireActual('./../../hooks/useFacebookAuth'),
  useFacebookAuth: jest.fn(),
}));

describe('Navigation component', () => {
  beforeEach(() => {
    useFacebookAuth.mockImplementation(() => ({
      authenticated: true,
      user: {
        name: 'test',
        picture: {
          data: {
            url: 'test',
          },
        },
      },
    }));
  });
  it('should be rendered', () => {
    const component = shallow(<Navigation />);
    expect(component).toMatchSnapshot();
  });

  it('should render user data', () => {
    const component = shallow(<Navigation />);
    expect(component).toMatchSnapshot();
  });
});
