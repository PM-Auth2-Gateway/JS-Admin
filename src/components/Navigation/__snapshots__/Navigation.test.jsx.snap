// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should be rendered 1`] = `
<Navbar
  className="border"
  collapseOnSelect={true}
  expand="md"
  variant="light"
>
  <NavLink
    to="/"
  >
    <NavbarBrand>
      <Image
        alt="icon"
        className="mr-2 align-center align-top"
        fluid={false}
        height="30"
        rounded={true}
        roundedCircle={false}
        src="/favicon.ico"
        thumbnail={false}
        width="30"
      />
    </NavbarBrand>
  </NavLink>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        activeClassName="text-primary"
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        to="/applications"
      >
        Applications
      </NavLink>
      <NavLink
        activeClassName="text-primary"
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        to="/guide"
      >
        Guide
      </NavLink>
      <NavLink
        activeClassName="text-primary"
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        to="/docs"
      >
        Docs
      </NavLink>
    </Nav>
    <Login />
  </NavbarCollapse>
</Navbar>
`;
